dbt/parser.py

```diff
@@ -341,57 +304,52 @@ def parse_schema_tests(tests, root_project, projects):
     return to_return
 
 
-def parse_schema_test(test_base, model_name, test_config, test_type,
-                      root_project_config, package_project_config,
-                      all_projects):
-    if test_type == 'not_null':
-        raw_sql = QUERY_VALIDATE_NOT_NULL.format(
-            ref="{{ref('"+model_name+"')}}", field=test_config)
-        name_key = test_config
+def get_nice_schema_test_name(test_type, test_name, args):
 
-    elif test_type == 'unique':
-        raw_sql = QUERY_VALIDATE_UNIQUE.format(
-            ref="{{ref('"+model_name+"')}}", field=test_config)
-        name_key = test_config
+    flat_args = []
+    for arg_name in sorted(args):
+        arg_val = args[arg_name]
 
-    elif test_type == 'relationships':
-        if not isinstance(test_config, dict):
-            return None
+        if type(arg_val) == dict:
+            parts = arg_val.values()
+        elif type(arg_val) in (list, tuple):
+            parts = arg_val
+        else:
+            parts = [arg_val]
```

Comment:
This func takes arbitrary schema test configs and spits out a human readable, unique name.

```yml
my_model:
  constraints:
    test_something:
      - { some_value: True, other_thing: ['abc', 'def'] }
```

So this func would operate on `{ some_value: True, other_thing: ['abc', 'def'] }` and spit out:

```
test_something_my_model_abc_def__True
```

I actually think that in practice this is kind of annoying/confusing. This "nice" name becomes the compiled filename and shows up in the `dbt test` output. Ideally, we'd keep the test config args around and show something like:

```
ERROR running test "test_something" for model "my_model" with args:
  some_value: True
  other_thing: ['abc', 'def']
```

but that's not really how things work currently. Something to consider for the future though